pkg/sbom/core/bom.go

```diff
@@ -269,6 +269,10 @@ func (b *BOM) AddRelationship(parent, child *Component, relationshipType Relatio
 	}
 }
 
+func (b *BOM) AddVulnerability(c *Component, vuln Vulnerability) {
```

Comment:
nit:
`AddVulnerability` and `AddVulnerabilities` use slightly different logic:
- AddVulnerability adds vulnerability to slice of existing vulnerabilities.
- AddVulnerabilities preserves vulnerabilities if component doesn't contain vulnerabilities. This behavior is more similar to `SetVulnerabilities`.

Reply:
Agree. Fixed in https://github.com/aquasecurity/trivy/pull/7340/commits/67961d7e7488e8292f43a5a1fabc41ad93aa7fad